client/src/slices/user.slice.ts

```diff
@@ -0,0 +1,184 @@
+      return thunkAPI.rejectWithValue(arg);
+    }
+    return arg;
+  });
+
+const register = createAsyncThunk(`${name}/register`,
+  async (arg: UserRegisterData, thunkAPI) => {
+    console.log('register', arg);
+
+    const userData = await UserClient.register({
+      firstName: arg.firstName,
+      lastName: arg.lastName,
+      phone: arg.phone,
+      email: arg.email,
+      service: arg.service,
+    });
+
+    const { localStorage } = thunkAPI.getState() as AppState;
+
+    return {
+      userData,
+      localStorage,
+    };
+  });
+
+const refresh = createAsyncThunk(`${name}/refresh`,
+  async (arg:void, thunkAPI) => {
+    const userData = await UserClient.current();
+
+    console.log('userData', userData);
+
+    const { localStorage } = thunkAPI.getState() as AppState;
+
+    return {
+      userData,
+      localStorage,
+    };
+  });
+
+export const userSlice = createSlice({
+  name,
+  initialState: userInitialState,
+  reducers: {
+    logout: () => {
+      return userInitialState;
+    },
+    changeOrg: (state, action: PayloadAction<{ orgId: number}>) => {
+      const orgId = action.payload.orgId;
+      const userRole = state.userRoles?.find(ur => ur.role.org?.id === orgId);
+      return {
+        ...state,
+        activeRole: userRole,
+      };
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(register.fulfilled, (state, action) => {
+      const { userData, localStorage } = action.payload;
+      const loggedInState = getLoggedInState(userData, localStorage);
+      return {
+        ...state,
+        ...loggedInState,
+      };
+    })
+      .addCase(refresh.fulfilled, (state, action) => {
+        const { userData, localStorage } = action.payload;
+        const loggedInState = getLoggedInState(userData, localStorage);
+        return {
+          ...state,
+          ...loggedInState,
+        };
+      })
+      .addCase(addFavoriteDashboard.fulfilled, (state, action) => {
+        const { workspaceId, dashboardUuid } = action.payload;
+        return addFavoriteDashboardImpl(state, workspaceId, dashboardUuid);
+      })
+      .addCase(addFavoriteDashboard.rejected, (state, action) => {
+        const { workspaceId, dashboardUuid } = action.payload as DashboardPayload;
+        return removeFavoriteDashboardImpl(state, workspaceId, dashboardUuid);
+      })
+      .addCase(removeFavoriteDashboard.fulfilled, (state, action) => {
+        const { workspaceId, dashboardUuid } = action.payload;
+        return removeFavoriteDashboardImpl(state, workspaceId, dashboardUuid);
+      })
+      .addCase(removeFavoriteDashboard.rejected, (state, action) => {
+        const { workspaceId, dashboardUuid } = action.payload as DashboardPayload;
+        return addFavoriteDashboardImpl(state, workspaceId, dashboardUuid);
+      })
+      .addDefaultCase(state => state);
+
+  },
+});
+
+function addFavoriteDashboardImpl(state: UserState, workspaceId: number, dashboardUuid: string) {
```

Comment:
These `Impl` functions can probably be simplified pretty nicely by leveraging immer.